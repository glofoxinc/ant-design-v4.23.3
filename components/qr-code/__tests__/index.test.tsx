import React from 'react';
import QrCode from '..';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';
import { render } from '../../../tests/utils';

describe('QrCode test', () => {
  mountTest(QrCode);
  rtlTest(QrCode);
  it('should correct render ', () => {
    const { container } = render(<QrCode value="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render `null` when value not exist', () => {
    const { container } = render(<QrCode value={undefined as unknown as string} />);
    expect(container.firstChild).toBe(null);
  });

  it('support icon', () => {
    const { container } = render(<QrCode value="test" icon="test" />);
    expect(container.querySelector<HTMLImageElement>('img')).toBeTruthy();
  });
  it('support size', () => {
    const { container } = render(<QrCode value="test" size={100} />);
    const wapper = container.querySelector<HTMLDivElement>('.ant-qrcode');
    expect(wapper?.style?.width).toBe('100px');
    expect(wapper?.style?.height).toBe('100px');
  });
});
