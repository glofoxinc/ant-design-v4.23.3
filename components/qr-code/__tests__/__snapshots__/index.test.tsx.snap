// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QrCode test rtl render component should be rendered correctly in RTL direction 1`] = `null`;

exports[`QrCode test should correct render  1`] = `
<div
  class="ant-qrcode"
  style="width: 128px; height: 128px;"
>
  <canvas
    height="128"
    style="height: 128px; width: 128px;"
    width="128"
  />
</div>
`;
